src/plugins/tools/apipTextToSpeech/plugin.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016-2019  (original work) Open Assessment Technologies SA;
+ *
+ * @author Anton Tsymuk <anton@taotesting.com>
+ */
+
+import _ from 'lodash';
+import __ from 'i18n';
+import 'ui/hider';
+import shortcut from 'util/shortcut';
+import namespaceHelper from 'util/namespace';
+import pluginFactory from 'taoTests/runner/plugin';
+import mapHelper from 'taoQtiTest/runner/helpers/map';
+
+const pluginName = 'apiptts';
+
+const actionPrefix = `tool-${pluginName}-`;
+
+/**
+ * Returns the configured plugin
+ */
+export default pluginFactory({
+    name: pluginName,
+
+    /**
+     * Initialize the plugin (called during runner's init)
+     */
+    init: function init() {
+        const testRunner = this.getTestRunner();
+        const testRunnerOptions = testRunner.getOptions();
+        const pluginShortcuts = (testRunnerOptions.shortcuts || {})[this.getName()] || {};
+
+        /**
+         * Checks if the plugin is currently available.
+         * To be activated with the special category x-tao-option-apiptts
+         *
+         * @returns {Boolean}
+         */
+        const isConfigured = () => mapHelper.hasItemCategory(
+            testRunner.getTestMap(),
+            testRunner.getTestContext().itemIdentifier,
+            'apiptts',
+            true
+        );
+
+        /**
+         * Is plugin activated ? if not, then we hide the plugin
+         */
+        const togglePlugin = () => {
+            if (isConfigured()) {
+                this.show();
+            } else {
+                this.hide();
+            }
+        };
+
+        /**
+         * Show the plugin panel
+         *
+         * @fires plugin-open.apiptts
+         */
+        const enablePlugin = () => {
+            this.button.turnOn();
+            this.setState('active', true);
+
+            testRunner.trigger(`plugin-open.${pluginName}`);
```

Comment:
no need to use this, you may directly use the plugin alias: `this.trigger('open')`

https://github.com/oat-sa/tao-core-sdk-fe/blob/master/src/core/plugin.js#L195